keras/engine/training.py

```diff
@@ -910,6 +910,8 @@ def _fit_loop(self, f, ins, out_labels=None, batch_size=32,
                         for l, o in zip(out_labels, val_outs):
                             epoch_logs['val_' + l] = o
             callbacks.on_epoch_end(epoch, epoch_logs)
+            if do_validation:
+                callbacks.on_evaluate_end(epoch, epoch_logs)
```

Comment:
Sorry for being obtuse, but I still see some sense in having on_evaluation_end(). 

In my specific case, I am trying to access the model history in a custom callback to write some graphs for internal states of the current epoch. It seems to me that keras updates the history after my callback is called, so when I try to access history.history["loss"] I actually get the loss of the epoch before the current one.

I admit, I don't know the best solution here. It certainly has merit to have the history updated after all other callbacks. On the other hand, adding an on_evaluate_end just for the history update may be a bit too much. Adding a flag somewhere that pushes a callback's execution to the very end may be an option, but I'm too fresh on the concepts of keras so I don't know what would be the cleanest solution here.